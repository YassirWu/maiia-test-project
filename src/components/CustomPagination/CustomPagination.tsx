import * as React from 'react';
import Pagination from "react-js-pagination";
import { makeStyles } from '@material-ui/styles'

const useStyles = makeStyles({
  root: {
    display: 'inline-block',
  },
  link: {
    padding: '6px 12px',
    textDecoration: 'none',
    color: '#337ab7',
    backgroundColor: '#fff',
    border: '1px solid #ddd',
  },
  activeLink: {
    color: '#fff',
    backgroundColor: '#337ab7',
    borderColor: '#337ab7',
    cursor: 'default',
  }
})

type CustomPaginationProps = {
  numberTotalElement: number;
  numberPerPage: number;
  currentPage: number;
  onChange: (page: number) => void;
}

const CustomPagination: React.FunctionComponent<CustomPaginationProps> = ({
  numberTotalElement,
  numberPerPage,
  currentPage,
  onChange,
}) => {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <Pagination
        activePage={currentPage}
        totalItemsCount={numberTotalElement}
        itemsCountPerPage={numberPerPage}
        pageRangeDisplayed={5}
        onChange={(page) => onChange(page)}
  
        prevPageText="<"
        nextPageText=">"
        linkClass={classes.link}
        activeLinkClass={classes.activeLink}
      />
    </div>
  );
};

export default CustomPagination;
