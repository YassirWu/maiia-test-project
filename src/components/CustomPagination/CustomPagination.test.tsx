import React from 'react';
import { shallow } from 'enzyme';
import CustomPagination from './CustomPagination';

describe('<CustomPagination />', () => {
  it('should render CustomPagination component', () => {

    const wrapper = shallow(
      <CustomPagination
        numberTotalElement={5000}
        numberPerPage={15}
        currentPage={2}
        onChange={jest.fn()}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
})


