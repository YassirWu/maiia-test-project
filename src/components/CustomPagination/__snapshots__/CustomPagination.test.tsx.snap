// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomPagination /> should render CustomPagination component 1`] = `ShallowWrapper {}`;
